src/main_loop.rs

```diff
@@ -811,13 +811,40 @@ impl MainLoopHandler {
 
                 self.main_to_miner_tx.send(MainToMiner::NewBlockProposal);
             }
+            PeerTaskToMain::DisconnectFromLongestLivedPeer => {
+                let global_state = self.global_state_lock.lock_guard().await;
+
+                // get all peers
+                let all_peers = global_state.net.peer_map.iter();
+
+                // filter out CLI peers
```

Comment:
Excellent that you're allowing CLI arguments to preserve peers that will never be disconnected.